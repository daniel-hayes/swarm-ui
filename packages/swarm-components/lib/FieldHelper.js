"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var FieldHelper = function FieldHelper(props) {
  return React.createElement("p", (0, _extends2["default"])({
    "data-swarm-field-helper": true
  }, props));
};

var _default = FieldHelper;
exports["default"] = _default;
FieldHelper.__docgenInfo = {
  "description": "",
  "methods": [],
  "displayName": "FieldHelper"
};