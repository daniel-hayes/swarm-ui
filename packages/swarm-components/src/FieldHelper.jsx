// @flow
import * as React from 'react';
import Icon from './Icon';

type Props = {};

const FieldHelper = (props: Props) => <p data-swarm-field-helper {...props} />

export default FieldHelper;