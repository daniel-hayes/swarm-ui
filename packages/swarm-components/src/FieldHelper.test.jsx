import React from 'react';

import { toMatchImageSnapshot } from 'jest-image-snapshot';
expect.extend({ toMatchImageSnapshot });

import getScreenRenderer from './testUtils/screenRenderer';

import FieldHelper from './FieldHelper';

describe('FieldHelper', () => {
	let renderer;

	// This is ran when the suite starts up.
	beforeAll(async () => {
		renderer = await getScreenRenderer({
			viewport: { width: 200, height: 100 },
			// verbose: true,
		});
	});

	// This is ran when the suite is done - stop the renderer.
	afterAll(() => {
		// comment next line out if you want to open it in your browser for debugging
		return renderer.stop();
	});
	it('Default', async () => {
		expect(
			await renderer.screenshot(<FieldHelper>Help me!</FieldHelper>)
		).toMatchImageSnapshot();
	});
	it('Custom styling', async () => {
		expect(
			await renderer.screenshot(
				<FieldHelper style={{ color: 'red' }}>
					Help me, I&apos;m red!
				</FieldHelper>
			)
		).toMatchImageSnapshot();
	});
});
