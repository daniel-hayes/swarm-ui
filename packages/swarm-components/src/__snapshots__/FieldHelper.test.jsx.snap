// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldHelper Snapshot: Custom Styling 1`] = `
<p
  data-swarm-field-helper={true}
  style={
    Object {
      "color": "red",
    }
  }
>
  Help me, I'm red!
</p>
`;

exports[`FieldHelper Snapshot: Default 1`] = `
<p
  data-swarm-field-helper={true}
>
  Help me!
</p>
`;
